### DISCUSSION TITLE
Revenue for the Django Debug Toolbar

### DISCUSSION THREAD

**Discussion by tim-schilling** [django-commons/django-debug-toolbar#2039] [Contributing]
Kicking this up again. It was pointed out to my that Tidelift would estimate the income for the Django Debug Toolbar at $100/mo. That combined with an Ethical Ads integration on ReadTheDocs ($30-50/mo), we're looking at $1500+ per year. That number is high enough for me to do something about it.

Unless someone has an idea where we need some funding for (maybe hosting an online demo site or a live version of something like the [Debug Tutorial](https://github.com/tim-schilling/debug-tutorial/)), I think we should route this back to the DSF.

Since I'm the one proposing it, I would volunteer to handle the finances and do some basic reporting of donations to the DSF here. I'm thinking a copy of the donation emails would suffice? If I notice a large impact on my taxes, we may need to revisit this, but I think this amount of money will be negligible.

So I would like discussion and (if you agree) approval from folks on three topics:

1. Should we setup Tidelift on the repo? This may result in additional work for us as Tidelift may require specific security measures to be implemented. I'm not sure what they are yet. I suspect it's the [security maintenance plan](https://tidelift.com/about/lifter-tasks#security) which would be good either way.
2. Should we setup Ethical Ads on ReadTheDocs?
3. Does anyone have any ideas on how to route the funds to the DSF? (I'll look into getting it directly to the DSF, but I suspect it has to go to a specific bank account)

cc @django-commons/django-debug-toolbar

**Comment by VeldaKiara** [comment]
On Tidelift its a yes because of security which is good for the toolbar, in case you need to help, ping me too. Would love to help or pair on it.

Ethical Ads is also a yes from me because of this:
> We’re building the advertising model we want to exist:
> We don’t track you
> We don’t sell your data
> We host everything ourselves, no third-party scripts or images

Basically everything is done by them. They also give back 10% to the community which is a plus on sustainability.

For funding on the DSF they use stripe so we may need to have a bank account card for the payment.

Side note: We could also set up a sponsor badge for the toolbar too and that could go to other open source programs in the Django ecosystem or conferences.

**Reply by VeldaKiara** [reply]
Although, I have a concern on your taxes, I think on the short term we could do that or probably there is a way that we account for taxes and cushion you too. So you can carry out that without worrying about taxes.

**Comment by elineda** [comment]
Hello, 

I'm ok for Tidelift but as DDT is not mean for production, they will accept it?
For Ad I'm ok for it. The demo site is a good idea. The only thing that worries me it's the security to host a debug app on a server.
For taxe two things: 
- Dunno in your country but mine donation to certains nonprofit organisation can lower your taxes. Maybe, it's the same for you.
- If it's to many, can you create a tiny charities ?

**Comment by tim-schilling** [comment]
It does appear that tidelift can [route money to organizations](https://support.tidelift.com/hc/en-us/articles/4406294816916-How-we-pay-lifters#h_01HGV73Q2G9XN5VVA2P933780Z). I think that could make this a bit easier to get on board with if we send the money directly to the DSF.

**Comment by tim-schilling** [comment]
@matthiask Can you let me know if you're good moving forward with Tidelift? There's an application process, but it requires me to sign an agreement. I'd rather not do that unless all four of us are on board. @elineda and @VeldaKiara were +1 on the idea, but had questions around the details.

**Comment by matthiask** [comment]
@tim-schilling yes, let's do it! I think it's a good idea.

**Comment by tim-schilling** [comment]
This has been started. I've got an account setup on https://server.ethicalads.io/ for ReadTheDocs

**Reply by tim-schilling** [reply]
I've also opened an account for thanks.dev. It appears that's one of the mechanisms that Sentry uses to donate to Open Source Software. That funding has been sitting waiting for us.

**Reply by VeldaKiara** [reply]
That's good news!

**Reply by tim-schilling** [reply]
I've also got Tidelift setup. This one is pointed at my bank account currently. I don't have auto-transfers set up. I want to see how close the estimate is to actuality. I also want to discuss with DSF and Tidelift if we can direct this at the DSF to avoid a tax obligation on my part, but we'll see.

If folks have questions, please ask. Happy to provide clarity where I can.

**Reply by tim-schilling** [reply]
Received our first payment from Tidelift!
### Receipt from Tidelift
![debug_toolbar_lift_202505](https://github.com/user-attachments/assets/9fc84c97-a16d-43f0-811e-89dc9d39990e)

### Donation proof
![Screenshot from 2025-05-08 07-32-22](https://github.com/user-attachments/assets/8eb7c8ef-3988-41ae-b9f9-0e16f9f87008)

**Reply by VeldaKiara** [reply]
OMGGG this is amazing. Open Source for the win.

**Reply by VeldaKiara** [reply]
Thank you @tim-schilling

**Reply by matthiask** [reply]
Thank you, that's awesome! @tim-schilling

**Reply by salty-ivy** [reply]
That's amazing !

**Reply by elineda** [reply]
thats great thank you @tim-schilling !

**Reply by tim-schilling** [reply]
<img width="530" alt="Screenshot 2025-06-22 at 6 25 42 PM" src="https://github.com/user-attachments/assets/15b85fbb-c05d-4113-8fe0-6b0ea01de191" />
<img width="792" alt="Screenshot 2025-06-22 at 6 12 12 PM" src="https://github.com/user-attachments/assets/40ba6977-3e4c-4f1e-93a2-e61868d65443" />
June's $100 payout from Tidelift and donation to the DSF.

I have set up a monthly payment to the DSF now.

**Reply by tim-schilling** [reply]
Thanks.dev payments from July
<img width="1285" height="501" alt="Screenshot from 2025-07-18 15-52-46" src="https://github.com/user-attachments/assets/84b9ff47-39fe-461f-974a-350015dcb2d8" />
<img width="771" height="222" alt="Screenshot from 2025-07-18 16-01-02" src="https://github.com/user-attachments/assets/bb920891-be04-43b2-9a16-ea1b042b6bb5" />

**Reply by tim-schilling** [reply]
August's data
<img width="1470" height="971" alt="2025-08-25 18 29 19 app pandadoc com e8011ec33516" src="https://github.com/user-attachments/assets/e0c3b197-4ff4-4b12-a300-821731168a52" />
<img width="987" height="644" alt="2025-08-25 18 34 00 mail google com e26faaee8b56" src="https://github.com/user-attachments/assets/5b159e92-aefc-4ae8-b030-eaabafd9c8d0" />


There's a problem with thanks.dev that we're trying to sort out. Until that's resolved, those amounts won't be transferred into my account, but they will still be accrued in thanks.dev

**Reply by tim-schilling** [reply]
Still pending the DSF donation (it's on a recurring schedule a few weeks later)
<img width="1463" height="986" alt="2025-09-14 18 57 37 app pandadoc com 28037923bfa4" src="https://github.com/user-attachments/assets/e6a87685-8042-4d38-994c-92e938546417" />

**Reply by tim-schilling** [reply]
<img width="525" height="366" alt="Screenshot from 2025-10-02 11-06-24" src="https://github.com/user-attachments/assets/bb3bc861-dbf9-475c-833a-9d883c0eb13e" />

**Reply by tim-schilling** [reply]
<img width="748" height="594" alt="Screenshot from 2025-11-05 14-17-10" src="https://github.com/user-attachments/assets/a3715719-6d9a-436e-bf9c-9abbd991d59d" />


<img width="490" height="314" alt="Screenshot from 2025-11-05 14-01-23" src="https://github.com/user-attachments/assets/996bbcb5-72b9-45be-a4a3-6cda8be6f1c5" />

<img width="748" height="594" alt="Screenshot from 2025-11-05 14-14-44" src="https://github.com/user-attachments/assets/cf9d60c7-1c2c-481a-9003-3f5bcf2728c7" />

**Reply by tim-schilling** [reply]
<img width="494" height="309" alt="DSF donation receipt November" src="https://github.com/user-attachments/assets/d4137ccc-086d-48c3-a60e-451c8c463b2a" />
<img width="831" height="667" alt="Tidelift payment November" src="https://github.com/user-attachments/assets/cbf2dc4a-8159-46a2-8d68-bd501b5ec4c6" />

**Reply by tim-schilling** [reply]
<img width="491" height="311" alt="Screenshot from 2026-01-13 13-53-05" src="https://github.com/user-attachments/assets/1cde152b-5448-4f1a-9976-297706747342" />
<img width="838" height="629" alt="Screenshot from 2026-01-13 13-53-43" src="https://github.com/user-attachments/assets/2b081866-3512-4b1e-ad71-ef8f7dc6896d" />

**Reply by tim-schilling** [reply]
<img width="544" height="353" alt="Screenshot from 2026-02-04 14-09-25" src="https://github.com/user-attachments/assets/f5d27dc9-678b-49eb-b4ea-7386658bc9e5" />
<img width="827" height="621" alt="Screenshot from 2026-02-04 14-10-01" src="https://github.com/user-attachments/assets/ebe2587c-8c1a-4bfa-8ac5-2b10781b4bbf" />

**Reply by tim-schilling** [reply]
<img width="618" height="261" alt="Screenshot from 2026-03-06 14-32-57" src="https://github.com/user-attachments/assets/f8c8f941-6f3b-4f0a-9489-a1ec65b53f11" />
<img width="614" height="512" alt="Screenshot from 2026-03-06 14-37-49" src="https://github.com/user-attachments/assets/ca6ae7dd-032f-4ff4-bb98-2b801867ef1f" />
<img width="832" height="594" alt="Screenshot from 2026-03-06 14-34-18" src="https://github.com/user-attachments/assets/557b1a5f-21af-44cd-bc2a-bad217c21ff0" />
<img width="495" height="321" alt="Screenshot from 2026-03-06 14-33-49" src="https://github.com/user-attachments/assets/85d40f38-ec0c-43ef-a882-46e215e8d3ea" />

I accumulated the thanks.dev funds from the past few months ($300) and donated that adhoc.

**Reply by tim-schilling** [reply]
Ethical Ads deposited $53.73 into the DSF's bank account directly in for revenue from October 1-March 31.

**Reply by tim-schilling** [reply]
Catherine from the DSF worked with me to change where the money from Tidelift goes. Rather than going into my bank account (where I need to claim it as income and get taxed for it), it now goes directly into the DSF's bank account. This will continue to be $100 per month. Since it's not longer hitting my account, I'm going to stop posting here about it.

I'll follow-up here whenever the thanks.dev deposits come through to my bank account and and I make a donation to the DSF.

**Reply by matthiask** [reply]
I'm glad you could get that sorted out.

**Comment by tim-schilling** [comment]
The Ethical Ads integration for the toolbar is now directly transferring the funds to the DSF via a Stripe integration.
